### PR TITLE
chore: streamline initialization flow - teleport - realm

### LINKED ISSUE #922: Jumping from a world loads you at 0,0 first before jumping
### **Explorer Alpha build version:**
Main

### **Issue Description:**
When jumping from world -> specific location (i.e.-150,0) it loads you at 0,0 first, and hides the loading screen - this could have a wider impact as LODs and scenes may begin loading in that initial location.

### **Steps to reproduce:**

  1. Jump to dcl experience
  2. Type /goto -150, 0

### PR DESCRIPTION
**Pull request by dalkia**
## What does this PR change?

Streamlines the teleport and realm change flow. Adapts the initialization flow to use this streamline.

- Removes teleport and landscape resposibilities from `RealUserInitializationFlowController` and moves them to `RealmNavigator`. Makes the loading process as simetrical as possible, making correct usage of loading reports. Nonetheless, they dont use the same method, since `RealUserInitializationFlowController` has extra tasks not needed by `RealmNavigator` (like `RealFlowLoadingStatus`, player loading and audio handling)
- Simplifies `AsyncLoadProcessReport`. Removes the async extension
- Unifies loading controller under `ILoadingScreen` interface
- Streamlines the realm, landscape and teleport logic in `RealmNavigator`. Further changes can be done in that class for elements that need to toggle on and off when leaving worlds.
- Now you can correctly teleport between realms. Landscape handling is done correctly, it can be further extended for reusability (not part of this PR)
- You can chose a realm on startup on editor
- When going into a realm, you will now spawn on the first fixed scene pointer of the realm; and not 0,0
- Comments out `CreateEmptyPointersInFixedRealmSystem` since we now have a landscape surrounding the world

Fixes #922, #892

Possibily fixes 919, but not referencing until we can confirm it.

image

## How to test the changes?

1. Launch the explorer
2. Go to goerli `/world goerli`. Check that the scene is loaded before loading screen is out
3. Go to genesis `/world genesis`. Check that the scene is loaded before loading screen is out
4. Teleport to several locations and look for irregularities
5. Go to Olavra `/world olavra.dcl.eth`. Check that the scene is loaded before loading screen is out
6. From Olavra, just teleport somewhere; '/goto 100,100' for example. Check that everything keeps working as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by anicalbano — APPROVED** [review]
🟢 Reviewed by QA

https://github.com/decentraland/unity-explorer/assets/139157886/bb886866-152d-4862-b69f-206d92c0e27f

File: Explorer/Assets/Scripts/Global/Dynamic/RealmNavigator.cs [hunk 1]
```diff
@@ -82,107 +94,188 @@ public async UniTask<bool> TryChangeRealmAsync(URLDomain realm, CancellationToke
...
-            await UniTask.WaitUntil(() => realmController.GlobalWorld.EcsWorld.Get<FixedScenePointers>(realmController.RealmEntity).AllPromisesResolved, cancellationToken: ct);
-
-            AssetPromise<SceneEntityDefinition, GetSceneDefinition>[] promises = realmController.GlobalWorld.EcsWorld.Get<FixedScenePointers>(realmController.RealmEntity).Promises;
+            AssetPromise<SceneEntityDefinition, GetSceneDefinition>[] promises = await WaitForFixedScenePromisesAsync(ct);
 
             var decodedParcelsAmount = 0;
 
             foreach (AssetPromise<SceneEntityDefinition, GetSceneDefinition> promise in promises)
+            {
+                //TODO (JUANI) : Shouldnt be possible, but some promises result are null
```
Comment (mikhail-dcl) [inline Explorer/Assets/Scripts/Global/Dynamic/RealmNavigator.cs:239]: Can we investigate it? I may cause soem other issues

File: Explorer/Assets/DCL/UserInAppInitializationFlow/RealUserInitializationFlowController.cs [hunk 2]
```diff
@@ -71,49 +57,46 @@ public class RealUserInitializationFlowController : IUserInAppInitializationFlow
...
+                = parentLoadReport.CreateChildReport(RealFlowLoadingStatus.PROGRESS[LandscapeLoaded]);
+            await realmNavigator.LoadTerrainAsync(landscapeLoadReport, ct);
+            parentLoadReport.SetProgress(loadingStatus.SetStage(LandscapeLoaded));
+
+            var teleportLoadReport
+                = parentLoadReport.CreateChildReport(RealFlowLoadingStatus.PROGRESS[PlayerTeleported]);
+            await realmNavigator.InitializeTeleportToSpawnPointAsync(teleportLoadReport, ct, startParcel);
+            parentLoadReport.SetProgress(loadingStatus.SetStage(PlayerTeleported));
+                    
+            parentLoadReport.SetProgress(loadingStatus.SetStage(Completed));
```
[the remark this answers is not in the document]
Comment (dalkia) [reply Explorer/Assets/DCL/UserInAppInitializationFlow/RealUserInitializationFlowController.cs:91]: Sure, why not

File: Explorer/Assets/Scripts/Global/Dynamic/RealmNavigator.cs (hunk 1 above)
Comment (dalkia) [reply Explorer/Assets/Scripts/Global/Dynamic/RealmNavigator.cs:239]: I wasnt able to reproduce it again, Im able to teleport back and forth from goerli with no problem. Probably I was missing an await when developing. I ll remove this code

Comment (mikhail-dcl) [reply Explorer/Assets/Scripts/Global/Dynamic/RealmNavigator.cs]: I tried to reproduce it but had no success either

**Pull request merged**